compiler/types.nim

```diff
@@ -496,9 +496,15 @@ proc typeToString(typ: PType, prefer: TPreferedDesc = preferName): string =
       add(result, typeToString(t.sons[i]))
     result.add "]"
   of tyAnd:
-    result = typeToString(t.sons[0]) & " and " & typeToString(t.sons[1])
+    for i in 0 .. t.sons.high:
```

Comment:
Minor nit, you can use `for i, typ in t.sons:` to get both the index and the element in a compact way